### PR TITLE
Implement freq_obs for sampling observations prior to optimization

### PR DESCRIPTION
**Pull request by dbrakenhoff**
# Short Description
This allows users to select a different time step for the observation data in `ml.solve()`. The model time step remains unchanged but the optimization is performed on a sample of the original oseries. E.g., model timestep is daily, but the model is fit on head observations once every 14 days:

```python
ml.solve(freq_obs="14D")
```

# Checklist before PR can be merged:
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [x] type hints for functions and methods
- [x] tests added / passed
- [x] Example (for new features)
- [x] Remove output for all notebooks with changes

### CODE REVIEW

**Comment by dbrakenhoff** [comment]
Test fails unrelated to this PR

File: pastas/model.py [hunk 1]
```diff
@@ -599,6 +600,7 @@ def observations(
         tmin: Optional[TimestampType] = None,
         tmax: Optional[TimestampType] = None,
         freq: Optional[str] = None,
+        freq_obs: Optional[str] = None,
```
Comment (raoulcollenteur) [inline pastas/model.py:603]: Do we need this here? Could this not be handled by the freq argument?

File: pastas/model.py [hunk 2]
```diff
@@ -741,6 +749,7 @@ def solve(
         initial: bool = True,
         weights: Optional[Series] = None,
         fit_constant: bool = True,
+        freq_obs: Optional[str] = None,
```
Comment (raoulcollenteur) [inline pastas/model.py:752]: Docstring is missing

File: pastas/model.py [hunk 3]
```diff
@@ -668,6 +673,7 @@ def initialize(
         weights: Optional[Series] = None,
         initial: bool = True,
         fit_constant: bool = True,
+        freq_obs: Optional[str] = None,
```
Comment (raoulcollenteur) [inline pastas/model.py:676]: Docstring is missing

**Review by raoulcollenteur — CHANGES_REQUESTED** [review]

**Comment by raoulcollenteur** [comment]
Thanks for the PR, this is a nice feature I think. Two things:

- Do we need the 'freq_obs' argument for the ml.observations? Can we need handle this through the 'freq' argument and only add freq_obs to ml.solve?
- Docstrings are missing for the new arguments.

ps. Update from dev-branch should solve CI issues :) 

Cheers,
Raoul

**Review by raoulcollenteur — APPROVED** [review]

**Review by raoulcollenteur — APPROVED** [review]

**Comment by raoulcollenteur** [comment]
Thanks for the last changes! Merging :)

**Pull request merged**